src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private double radius;
+
+    public Circle(Color color, double radius) {
+        super(color);
+        this.radius = radius;
+    }
+
+    @Override
+    public double calculateArea() {
+        return Math.PI * Math.pow(radius, 2);
+    }
+
+    @Override
+    public void drawFigure() {
+        System.out.println("Figure: circle, area: " + calculateArea() + " sq. units, radius: "
+                + getRadius() + " units, color: " + getColor());
```

Comment:
The `drawFigure()` method mixes concerns by both calculating the area and handling output formatting. It would be better to separate these concerns for cleaner code.